Revamp accordion borders to fix pixel jumping

mdo opened the pull request:
This _should_ fix the issue once and for all—this takes a simpler approach to the borders and seems to fix all the movement I could see (which was actually visible without even zooming in, at least in Safari).

- Pushes the borders to the `.accordion-item` instead of the `.accordion-button`
- Simplifies the `.accordion-flush`'s `border` resets
- Recreates the `.accordion-button`'s `border-bottom` with an inset `box-shadow`
- Moves the `border-radius` to the parent, which necessitated the new `$accordion-inner-border-radius` variable

Fixes #32250, closes #32251.

Preview: https://deploy-preview-33149--twbs-bootstrap.netlify.app/docs/5.0/components/accordion/

XhmikosR commented:
Definitely better from a quick look!

It seems there's some new issues, like double border in the first item, missing border-bottom is the other items, and a slight border-radius issue in the last item

![image](https://user-images.githubusercontent.com/349621/108470396-95443900-7292-11eb-9877-f5fa5c775c9c.png)

mdo commented:
Pushed an update that I hope will address the FF zoom issue.

mdo committed:
Revamp accordion borders to fix pixel jumping when toggling

- Pushes the borders to the .accordion-item
- Simplifies the .accordion-flush
- Recreates the .accordion-button border-bottom with an inset box-shadow
- Updates background-color to solid to match other components like list group and prevent stacked translucent borders
- Use negative margin to overlap the consecutive items

mdo commented:
@ffoodd Any chance you could give me a quick review here? <3

ffoodd commented:
I still see a little glitch but it's better.

FF zoom issue persists (I start to see weirdness from 110% to 170%, either a missing border or doubled ones).

I guess our main issue relates more to animating `height` than to borders. Shouldn't we try something based on `visibility` & `max-height` instead? My main reference on the topic is [this van11y hide/show plugin](https://van11y.net/accessible-hide-show/) (check the "Animation" section in "Bonuses").

Not quite sure this is the way to go, though…

mdo commented:
@ffoodd I think we merge this as-is and then look at revamping the animations separately. Thoughts?

ffoodd commented:
Agreed, :shipit:

Should probably open a new issue dedicated to animation though.

ffoodd commented on scss/_accordion.scss:
```diff
@@ -12,18 +12,15 @@
   &:not(.collapsed) {
     color: $accordion-button-active-color;
     background-color: $accordion-button-active-bg;
+    box-shadow: inset 0 -1px 0 $accordion-border-color;
```
Shouldn't we use a variable for `1px` here?

ffoodd reviewed (approved):
Only left a question. Simpler and better solution IMHO, let's do this.

mdo committed:
Use variable

mdo committed:
make linter happy, multiple by -1
